modules/compliance-custom-node-pools.adoc

```diff
@@ -0,0 +1,93 @@
+
+:_content-type: PROCEDURE
+[id="compliance-custom-node-pools"]
+= Creating custom node pools
+
+The Compliance Operator does not maintain a copy of each node pool configuration. The Compliance Operator aggregates consistent configuration options for all nodes within a single node pool into one copy of the configuration file. The Compliance Operator then uses the configuration file for a particular node pool to evaluate rules against nodes within that pool.
+
+If your cluster uses custom node pools outside the default `worker` and `master` node pools, you will need to supply additional variables to ensure the Compliance Operator aggregates a configuration file for that node pool.
+
+.Procedure
+
+. To check configuration against all pools in an example cluster containing `master`, `worker`, and `infra` node pools, set the value of `ocp-var-role-master` and `opc-var-role-worker` to `infra` in the `TailoredProfile`:
++
+[source,yaml]
+----
+apiVersion: compliance.openshift.io/v1alpha1
+kind: TailoredProfile
+metadata:
+  name: cis-infra-tp
+spec:
+  extends: ocp4-cis
+  title: My modified nist profile with a custom value
+  setValues:
+  - name: ocp4-var-role-master
+ value: infra
+ rationale: test for infra nodes
+  - name: ocp4-var-role-worker
+ value: infra
+ rationale: test for infra nodes
+  description: cis-infra-scan
+----
+
+. Add the `infra` role to the `ScanSetting` that will be stored in the `ScanSettingBinding`:
++
+[source,yaml]
+----
+apiVersion: compliance.openshift.io/v1alpha1
+kind: ScanSetting
+metadata:
+  name: default
+  namespace: openshift-compliance
+rawResultStorage:
+  rotation: 3
+  size: 1Gi
+roles:
+- worker
+- master
+- infra
+scanTolerations:
+- effect: NoSchedule
+  key: node-role.kubernetes.io/master
+  operator: Exists
+schedule: '0 1 * * *'
+----
+
+. Create a scan using `ScanSettingBinding`:
++
+[source,yaml]
+----
+apiVersion: compliance.openshift.io/v1alpha1
+kind: ScanSettingBinding
+metadata:
+  name: cis
+  namespace: openshift-compliance
+profiles:
+- apiGroup: compliance.openshift.io/v1alpha1
+  kind: Profile
+  name: ocp4-cis
+- apiGroup: compliance.openshift.io/v1alpha1
+  kind: Profile
+  name: ocp4-cis-node
+- apiGroup: compliance.openshift.io/v1alpha1
+  kind: TailoredProfile
+  name: cis-infra-tp
+settingsRef:
+  apiGroup: compliance.openshift.io/v1alpha1
+  kind: ScanSetting
+  name: default
+----
+
+The Compliance Operator checks the runtime `KubeletConfig` through the `Node/Proxy` API object, and then uses variables such as `ocp-var-role-master` and `ocp-var-role-worker` to determine the nodes it performs the check against. In the `ComplianceCheckResult`, the `KubeletConfig` rules will be shown as `ocp4-cis-kubelet-*`. The scan only passes if all selected nodes pass this check.
+
+.Verification
+
+* You can verify if a rule is checked through the `Node/Proxy` object by checking if `valuesUsed` of the `ComplianceCheckResult` contains `ocp4-var-role-master` or `ocp4-var-role-worker`:
```

Comment:
Had a second look on this, I think this is invalid, can we change this to 

All the Platform KubeletConfig rules are checked through the `Node/Proxy` object, you can find those rules by:
```
oc get rules -o json | jq '.items[] | select(.checkType == "Platform") | select(.metadata.name | contains("ocp4-kubelet-")) | .metadata.name'
```